modules/http/http-worker.c

```diff
@@ -419,7 +441,11 @@ _flush_on_target(HTTPDestinationWorker *self, HTTPLoadBalancerTarget *target)
   if (http_code == 401 && owner->auth_header)
     return _renew_header(owner);
 
-  return map_http_status_to_worker_status(self, target->url, http_code);
+  HttpResponseHandler *response_handler = NULL;
+  if ((response_handler = http_response_handlers_lookup(owner->response_handlers, http_code)))
+    return _map_http_result(response_handler->action(response_handler->user_data));
+
+  return default_map_http_status_to_worker_status(self, target->url, http_code);
```

Comment:
It would be even more readable if this would be factored out as a separate `map_http_result` function.
something like similar:

```
static LogThreadedResult
_try_to_custom_map_http_status_to_worker_status(HttpDestinationWorker *self, glong http_code)
{
  HttpResponseHandler *response_handler = NULL;

  if ((response_handler = http_response_handlers_lookup(owner->response_handlers, http_code)))
    return _custom_map_http_result(response_handler->action(response_handler->user_data));

  return LTR_MAX;
}
static LogThreadedResult
_map_http_status_code(HttpDestinationWorker *self, const gchar *url, glong http_code)
{
  LogThreadedResult result = _try_to_custom_map_http_status_to_worker_status(self, http_code);
  
  if (result != LTR_MAX)
      return result;
  
  return default_map_http_status_to_worker_status(self, url, http_code);
}
```

Reply:
Fixed.